edge-util/src/Microsoft.Azure.Devices.Edge.Storage.RocksDb/disk/StorageSpaceChecker.cs

```diff
@@ -0,0 +1,149 @@
+            Option<string> driveName = drive.Map(d => d.Name);
+            StorageSpaceCheckerBase inner = driveName.Map(d => new ThresholdStorageSpaceChecker(d, thresholdPercentage, checkFrequency, Events.Log) as StorageSpaceCheckerBase)
+                .GetOrElse(() => new NullStorageSpaceChecker(checkFrequency, Events.Log));
+            var diskSpaceChecker = new StorageSpaceChecker(storageFolder, driveName, checkFrequency, inner);
+            Events.Created(storageFolder);
+            return diskSpaceChecker;
+        }
+
+        public void SetThresholdPercentage(int thresholdPercentage)
+        {
+            lock (this.updateLock)
+            {
+                this.drive.ForEach(d =>
+                {
+                    Events.SetMaxPercentageUsage(thresholdPercentage, d);
+                    this.inner = new ThresholdStorageSpaceChecker(d, thresholdPercentage, this.checkFrequency, Events.Log);
+                });
+            }
+        }
+
+        public void SetMaxDiskUsageSize(long maxDiskUsageBytes)
+        {
+            lock (this.updateLock)
+            {
+                Events.SetMaxSizeDiskSpaceUsage(maxDiskUsageBytes, this.storageFolder);
+                this.inner = new FixedSizeSpaceChecker(this.storageFolder, maxDiskUsageBytes, this.checkFrequency, Events.Log);
+            }
+        }
+
+        internal static Option<DriveInfo> GetMatchingDrive(string storageFolder)
+        {
+            DriveInfo match = null;
+            var drives = new DriveInfo[0];
+            try
+            {
+                drives = DriveInfo.GetDrives();
+                int segmentCount = -1;
+                foreach (DriveInfo drive in drives)
+                {
+                    if (drive.IsReady)
+                    {
+                        if (storageFolder.StartsWith(drive.Name, StringComparison.OrdinalIgnoreCase))
+                        {
+                            string[] segments = drive.Name.Split(new[] { Path.DirectorySeparatorChar }, StringSplitOptions.RemoveEmptyEntries);
+                            if (segments.Length > segmentCount)
+                            {
+                                segmentCount = segments.Length;
+                                match = drive;
+                            }
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Events.ErrorGettingMatchingDrive(storageFolder, e);
+            }
+
+            if (match == null)
+            {
+                Events.NoMatchingDriveFound(drives, storageFolder);
+                return Option.None<DriveInfo>();
+            }
+
+            Events.FoundDrive(storageFolder, match);
+            return Option.Maybe(match);
+        }
+
+        static class Events
+        {
+            public static readonly ILogger Log = Logger.Factory.CreateLogger<StorageSpaceChecker>();
+            const int IdStart = 50000;
```

Comment:
>50000 [](start = 32, length = 5)

Might want to declare this in here - edge-util\src\Microsoft.Azure.Devices.Edge.Util\UtilEventsIds.cs